app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -32,28 +33,41 @@ import uk.gov.hmrc.securemessage.controllers.model.common.read._
 import uk.gov.hmrc.securemessage.controllers.model.common.write._
 import uk.gov.hmrc.securemessage.controllers.utils.EnrolmentHelper._
 import uk.gov.hmrc.securemessage.controllers.utils.QueryStringValidation
-import uk.gov.hmrc.securemessage.models.core.Conversation
-import uk.gov.hmrc.securemessage.services.SecureMessageService
+import uk.gov.hmrc.securemessage.services.{ Auditing, ErrorHandling, SecureMessageService }
 import uk.gov.hmrc.time.DateTimeUtils
 
-import javax.inject.Inject
 import scala.concurrent.{ ExecutionContext, Future }
 
 @SuppressWarnings(Array("org.wartremover.warts.ImplicitParameter"))
 class SecureMessageController @Inject()(
   cc: ControllerComponents,
   val authConnector: AuthConnector,
+  override val auditConnector: AuditConnector,
   secureMessageService: SecureMessageService,
   dataTimeUtils: DateTimeUtils)(implicit ec: ExecutionContext)
-    extends BackendController(cc) with AuthorisedFunctions with QueryStringValidation with I18nSupport with Logging {
+    extends BackendController(cc) with AuthorisedFunctions with QueryStringValidation with I18nSupport
+    with ErrorHandling with Auditing {
 
   def createConversation(client: ClientName, conversationId: String): Action[JsValue] = Action.async(parse.json) {
     implicit request =>
       client match {
         case ClientName.CDCM =>
           withJsonBody[CdcmConversation] { cdcmConversation =>
-            saveConversation(
-              cdcmConversation.asConversationWithCreatedDate(client.entryName, conversationId, dataTimeUtils.now))
+            val conversation =
+              cdcmConversation.asConversationWithCreatedDate(client.entryName, conversationId, dataTimeUtils.now)
+            secureMessageService
+              .createConversation(conversation)
+              .map {
```

Comment:
`Future` already wraps `Try` which can contain an exception with a stacktrace. Why do we re-wrap in `Either` if the left side still extends `RuntimeException`? Not sure if this was there already, but worth the question.

Reply:
Presumably this is because the `Left` of `Either` returns a `SecureMessageError` rather than a general `Exception` but perhaps @OlegEfrem can answer this as he wrote it :)

Reply:
@rwalpole That should make no difference, assuming the `SecureMessageError` is an ADT that doesn't derive from `Exception` at all. @OlegEfrem can answer hopefully.

Reply:
@alexflav23 initially we had an ADT not exteding exception, but then we realised there might be also exceptions thrown in the libraries which are outside our control which we would want to hand in the error handling function. This is the reason we extended our ADT error from Exception (not throwable, as throwable would include Error which is fatal, Exception is NonFatal). 
The reason why we have Either instead of Try/Future is that we want to have intention revealing functions with specialised exeption (the top of the Left would be SecureMessageError, never any other errors).